service/src/io/pedestal/http/resources/impl.clj

```diff
@@ -0,0 +1,149 @@
+; Copyright 2025 Nubank NA
+
+; The use and distribution terms for this software are covered by the
+; Eclipse Public License 1.0 (http://opensource.org/licenses/eclipse-1.0)
+; which can be found in the file epl-v10.html at the root of this distribution.
+;
+; By using this software in any fashion, you are agreeing to be bound by
+; the terms of this license.
+;
+; You must not remove this notice, or any other, from this software.
+
+(ns ^:no-doc io.pedestal.http.resources.impl
+  "Implementation details for io.pedestal.http.resource; subject to change at any time."
+  (:require [clojure.java.io :as io]
+            [clojure.string :as string]
+            [ring.util.io :as util.io])
+  (:import (jakarta.servlet.http HttpServletResponse)
+           (java.io BufferedInputStream File InputStream)
+           (java.net URL)
+           (java.nio.channels Channels FileChannel)
+           (java.nio.file OpenOption StandardOpenOption)
+           (java.util Date)
+           (java.util.jar JarFile)))
+
+;; This adapts some of the ideas from ring.util.response
+
+(def ^:private open-options
+  (into-array OpenOption [StandardOpenOption/READ]))
+
+(defn- should-stream?
+  [request stream-size]
+  (let [{:keys [servlet-response]} request
+        buffer-size-bytes (if servlet-response
+                            (.getBufferSize ^HttpServletResponse servlet-response)
+                            ;; The comment: Assume 1500 MTU (https://en.wikipedia.org/wiki/Maximum_transmission_unit)
+                            ;; (originally copied from io.pedestal.http.ring-middlewares/fast-resource)
+                            ;; Q: Then why is the default 1460?
```

Comment:
Probably to account for 20 bytes for TCP header + 20 bytes for IP header (assuming IPv4)